### ISSUE TITLE
Android Tests run slow

### ISSUE THREAD

**Issue by fknives** [fknives/Android-Tutorial-Test-ShowCase#39]
**Describe the bug**
Android tests run unnecessarily slow

**To Reproduce**
Running the tests will show that Espresso is waiting a lot, because an activity is not resumed.

**Expected behavior**
The tests shouldn't take so long to run

**Smartphone (please complete the following information):**
 - Device: doesn't matter
 - OS: doesn't matter

**Additional context**
Seems to be caused by the idling and synchronizations of AndroidTests. Since the Dispatcher is not swapped, when synchronizing with requests it seems to be stoping by waiting on the activities root.

**Comment by fknives** [comment]
It seems to be waiting for the Activity since the ActivityLifecycleMonitorRegistry still listens to the activity's callback.
That's because instead of activityScenario.close() we just destroy the activity.

**Comment by fknives** [comment]
Reopening because of the issue Splash activity happy flow taking long.

However removing milestone, since it's now lower priority

**Comment by fknives** [comment]
The issue seems to be related to android-test/issues/676

Since it actually hangs when the activityScenario.moveToState(Lifecycle.State.DESTROYED) / activityScenario.close() is called
